### PR TITLE
fix(date-picker): Allow changing year in header when minValue/maxValue are set.

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #4186

## Summary

fix(date-picker): can't change year when using min/max props #4186

### CODE REVIEW

File: src/components/date-picker-month-header/date-picker-month-header.tsx [hunk 1]
```diff
@@ -259,27 +264,50 @@ export class DatePickerMonthHeader {
...
-   * set to active if in range
-   */
-  private setYear(localizedYear: string, increment = 0) {
-    const { min, max, activeDate, localeData, yearInput } = this;
+  private getInRangeDate({
+    localizedYear,
+    increment = 0
+  }: {
+    localizedYear: string;
+    increment?: number;
```
Comment (jcfranco) [inline src/components/date-picker-month-header/date-picker-month-header.tsx:272]: Nitpick: is there another name we can use for this var? `increment: -1` threw me off. Same for `setYear`.

File: src/components/date-picker-month-header/date-picker-month-header.tsx [hunk 2]
```diff
@@ -259,27 +264,50 @@ export class DatePickerMonthHeader {
...
+    // if you've supplied a year and it's in range, update active date
+    if (inRangeDate) {
       this.calciteDatePickerSelect.emit(inRangeDate);
-      yearInput.value = localizeNumber(inRangeDate.getFullYear(), localeData);
-    } else {
-      // leave the current active date and clean up garbage input
-      yearInput.value = localizeNumber(activeDate.getFullYear(), localeData);
+    }
+
+    if (reset) {
```
Comment (jcfranco) [inline src/components/date-picker-month-header/date-picker-month-header.tsx:309]: Nitpick: in terms of naming, is this behavior more of a `commit` than a `reset`?

**Review by jcfranco — APPROVED** [review]
Awesome! Before merging, can you add a test for this?

**Review by jcfranco — COMMENTED** [review]
Can you also reword the commit title to describe the fix vs the bug?

File: src/components/date-picker-month-header/date-picker-month-header.tsx (hunk 1 above)
Comment (driskull) [reply src/components/date-picker-month-header/date-picker-month-header.tsx:272]: Yeah, not sure on this one. maybe offset? I'll change to that for now.

**Pull request merged**